types/react-redux-epic/index.d.ts

```diff
@@ -0,0 +1,22 @@
+// Type definitions for react-redux-epic 1.1
+// Project: https://github.com/BerkeleyTrue/react-redux-epic#readme
+// Definitions by: forabi <https://github.com/forabi>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.6
+
+import * as React from 'react';
+import { Observable } from 'rxjs/Observable';
+import { Epic } from 'redux-observable';
+
+export interface Action {
+    type: string;
+}
+
+export function wrapRootEpic<T extends Action, S, D = any>(
+    epic: Epic<T, S, D> // tslint:disable-line no-unnecessary-generics
```

Comment:
Better way to fix the lint would be to just return `: Epic<T, S, D>` instead of `typeof epic`, so it's clear the type parameters are used more than once.

Reply:
Would TypeScript still be able to infer that the wrapped root epic takes exactly the same type of actions? My use case includes custom action types, and my store creator function expects that the epic is compatible with the defined types only.

Reply:
It should, but I would add a test to make sure.

Reply:
I just checked, it looks like it does not, even with `typeof epic`. So I added the type parameters back and used it them again in the return type. This should solve both issues :smile: